### PR TITLE
Sk0 51

### PR DESCRIPTION
**Pull request by Thomas851204**
Authguard on FE completed. Requesting code review.

### CODE REVIEW

**Comment by markkovari** [comment]
Screenshot 2024-02-12 at 20 40 11
Hey @Thomas851204 you have a really good question, although there is no "simple" answer unfortunately.:| 
An other alternative: https://stackoverflow.com/questions/47026500/angular-guard-routes-by-default
but programmatically we can extend objects with a `map`

File: sk0-front/src/app/guards/auth.guard.ts
Comment (markkovari) [inline sk0-front/src/app/guards/auth.guard.ts]: Unfortunately this variable is not reactive. It should be an Observable or Promise.

Comment (markkovari) [reply sk0-front/src/app/guards/auth.guard.ts]: if the `isUserLoggedIn$` is a `Subject` then it *might* be only 
```ts
export const authGuard: CanActivateFn = () => inject(AuthService).isUserLoggedIn$
```

Comment (Thomas851204) [reply sk0-front/src/app/guards/auth.guard.ts]: Thats fair. I saw that this variable is changed to Behaviorsubject in the authService login and register functions PR. Should I wait with the correction until that branch is merged into dev, or should I cherry-pick those commits into this branch?

Comment (markkovari) [reply sk0-front/src/app/guards/auth.guard.ts]: Well I would merge it like this. My commits are most of the times rather guidance than requirements. We should fix it when we migrate to reactive state management (subjects/observables).

**Review by markkovari — APPROVED** [review]
Just one nit-picking move: feel free to use the backlog to accumulate technical debt too. (comments in code are not the best transparent places unfortunately)

**Comment by Thomas851204** [comment]
Need another review due to changes after merge conflict resolution.

**Review by rolcho — APPROVED** [review]
Working for me too

**Review by bendesbeus — APPROVED** [review]

**Pull request merged**
